pkg/operator/mon/health.go

```diff
@@ -19,15 +19,49 @@ package mon
 
 import (
 	"fmt"
+	"time"
 
 	"github.com/rook/rook/pkg/ceph/client"
 	"github.com/rook/rook/pkg/ceph/mon"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// CheckHealth for the monitors
-func (c *Cluster) CheckHealth() error {
+var (
+	healthCheckInterval = 10 * time.Second
+)
+
+// HealthChecker check health for the monitors
+type HealthChecker struct {
+	monCluster *Cluster
+}
+
+// NewHealthChecker creates a new HealthChecker object
+func NewHealthChecker(monCluster *Cluster) *HealthChecker {
+	return &HealthChecker{
+		monCluster: monCluster,
+	}
+}
+
+// Check periodically the health of the monitors
+func (hc *HealthChecker) Check(stopCh chan struct{}) {
+	for {
+		select {
+		case <-stopCh:
+			logger.Infof("Stopping monitoring of cluster in namespace %s", hc.monCluster.Namespace)
+			return
+
+		case <-time.After(healthCheckInterval):
+			logger.Debugf("checking health of mons")
+			err := hc.monCluster.checkHealth()
+			if err != nil {
+				logger.Infof("failed to check mon health. %+v", err)
+			}
+		}
+	}
+}
+
+func (c *Cluster) checkHealth() error {
```

Comment:
I thought about it but i concluded that CheckHealth could be made generic or an interface so that a controller like HealthChecker can check the health of many resources like mon, osd, pools, rgw etc. Right now we are only using it for mon.

Either way i am also ok to make it a type HealthCheck instead